index.js

```diff
@@ -573,8 +574,23 @@ class Raspi extends EventEmitter {
     throw new Error('reportDigitalPin is not yet implemented');
   }
 
-  pingRead() {
-    throw new Error('pingRead is not yet implemented');
+  pingRead(config, handler) {
+    const pulseOut = config.pulseOut;
+    const pin = config.pin;
+    const sonar = new Sonar(config.pin);
+    
+    const pinInstance = this[getPinInstance](this.normalize(pin));
+    if (pinInstance.mode != INPUT_MODE) {
+      this.pinMode(pin, INPUT_MODE);
+    }
+    
+    const interval = setInterval(() => {
+      sonar.read(handler);
+    }, pulseOut);
```

Comment:
@rwaldron the algorithm was translated from a python example. My thoughts also were that that the constant polling would tie up the CPU; the plan is to switch using interrupts in raspi-sonar once integration is completed. Would you see this as a more viable approach? Especially for using more than a single ultrasonic sensor?